genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/MonitorJobAction.java

```diff
@@ -72,6 +95,16 @@ protected Events executeStateAction(final ExecutionContext executionContext) {
 
         log.info("Job process completed with final status {}", finalJobStatus);
 
+        final File jobDirectory = executionContext.getJobDirectory().get();
+        final JobSpecification jobSpecification = executionContext.getJobSpecification().get();
+
+        // Cleanup job directory before archiving it
+        cleanupJobDirectory(jobDirectory);
+
+        // Archive job outputs before setting the final (non-active) status.
```

Comment:
This works for now. We may want to consider introducing an intermediate state like `CLEANUP` that would allow clients to know the actual job execution is done but reflect Genie isn't done with everything? Not part of this commit just a thought